File: src/com/openc2e/plugins/intellij/caos/def/lang/CaosDefFileType.java
package com.openc2e.plugins.intellij.caos.def.lang;

import com.intellij.openapi.fileTypes.LanguageFileType;
import icons.CaosScriptIcons;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class CaosDefFileType extends LanguageFileType {

    @NonNls
    public static final CaosDefFileType INSTANCE = new CaosDefFileType();
    @NonNls
    public static final String DEFAULT_EXTENSION = "caosdef";
    @NonNls
    public static final String DOT_DEFAULT_EXTENSION = "."+DEFAULT_EXTENSION;

    private CaosDefFileType() {
        super(CaosDefLanguage.Companion.getInstance());
    }

    @NotNull
    @Override
    public String getName() {
        return "Caos Definitions File";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "A Caos definitions file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return DEFAULT_EXTENSION;
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return CaosScriptIcons.CAOS_DEF_FILE_ICON;
    }
}

File: src/main/java/com/badahori/creatures/plugins/intellij/agenteering/caos/def/lang/CaosDefFileType.kt
@file:Suppress("unused")

package com.badahori.creatures.plugins.intellij.agenteering.caos.def.lang

import com.badahori.creatures.plugins.intellij.agenteering.caos.def.lang.CaosDefLanguage.Companion.instance
import com.intellij.openapi.fileTypes.LanguageFileType
import icons.CaosScriptIcons
import org.jetbrains.annotations.NonNls
import javax.swing.Icon


object CaosDefFileType : LanguageFileType(instance) {
    override fun getName(): String {
        return "Caos Definitions File"
    }

    override fun getDescription(): String {
        return "A Caos definitions file"
    }

    override fun getDefaultExtension(): String {
        return DEFAULT_EXTENSION
    }

    override fun getIcon(): Icon? {
        return CaosScriptIcons.CAOS_DEF_FILE_ICON
    }
        @NonNls
        val DEFAULT_EXTENSION = "caosdef"

        @NonNls
        val DOT_DEFAULT_EXTENSION = ".$DEFAULT_EXTENSION"
}